src/main/java/seedu/address/model/EntityList/MentorList.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.model.EntityList;
+
+import java.util.ArrayList;
+import java.util.List;
+import seedu.address.model.Entity.Entity;
+import seedu.address.model.Entity.ID;
+import seedu.address.model.Entity.Mentor;
+import seedu.address.model.Entity.PrefixType;
+
+public class MentorList extends EntityList {
+    private List<Mentor> mentors;
+
+    /**
+     * Constructor.
+     *
+     */
+    public MentorList() {
+        super();
+        this.mentors = new ArrayList<>();
+    }
+
+    /**
+     * Gets Mentor by ID.
+     *
+     * @param id
+     * @return
+     */
+    @Override
+    public Mentor get(ID id) {
+        // TODO
+        return null;
+    }
+
+    /**
+     * Updates Mentor by ID.
+     *
+     * @param mentor
+     * @throws Exception if error while updating
+     */
+    @Override
+    public void update(Entity mentor) throws Exception {
+        // TODO
+        mentors.add((Mentor) mentor);
+    }
+
+    /**
+     * Adds the mentor into the list
+     *
+     * @param mentor
+     * @throws Exception
+     */
+    @Override
+    public void add(Entity mentor) throws Exception {
+        mentors.add((Mentor) mentor);
+    }
+
+    /**
+     * Deletes team by id.
+     *
+     * @param id
+     * @throws Exception
+     */
+    @Override
+    public void delete(ID id) throws Exception {
+        for (Mentor m: this.mentors) {
+            if (m.getId() == id) {
+                this.mentors.remove(m);
+                return;
+            }
+        }
+    }
+
+    /**
+     * List the mentors.
+     *
+     * @return List<Mentor>
+     */
+    @Override
+    public List<? extends Entity> list() {
```

Comment:
Maybe i'm mistaken, but should the return type of this be List<Mentor> instead?

Reply:
Don't quite get, what do you mean by List

Reply:
Sorry I think the type got cut out as HTML instead. 

I meant should the return type be List < Mentor > instead? Because right now it returns List < ? extends Entity >.

Reply:
Ah but then it wont override the abstract method in the parent